Parallel import

tristan0x opened the pull request.

pramodk commented:
@smelchio : does this include fixes that you were mentioning on thur/friday? (by the way, you can merge/rebase master branch to resolve conflict).

tristan0x commented on src/Omega_h_gmsh.cpp:
```diff
@@ -581,6 +732,49 @@ void write(filesystem::path const& filepath, Mesh* mesh) {
+  const auto coords = mesh.coords();
+
+  std::vector<std::size_t> node_tags(static_cast<size_t>(nnodes));
+  std::vector<double> node_coords(3*static_cast<size_t>(nnodes), 0);
```
Can you please confirm @smelchio  that the hard-coded 3 is normal because gmsh is always writing nodes in 3d?
If so I would replace all occurences of 3 by something a bit more explicit, for instance:

`const auto gmsh_dims = 3;`

smelchio replied:
I confirm that gmsh always have 3 coordinates even for a 1D or a 2D mesh. If you are not in 3D, the z coordinate is not optional, i.e., coordinates are always a triplet (x, y , z) in gmsh (see e.g., http://gmsh.info/doc/texinfo/gmsh.html#File-formats )

Commit:
Support Gmsh parallel format using Gmsh C++ API

Provides 2 new functions in Omega_h API:
* Omega_h::gmsh::read_parallel
* Omega_h::gmsh::write_parallel

Co-authored-by: Samuel Melchior <35073063+smelchio@users.noreply.github.com>

Commit:
Fix loading of physical entities in Gmsh loader

Store the physical entities in Mesh::class_sets
instead of the local element identifiers and tag the elements
with their corresponding entities.

Co-authored-by: Samuel Melchior <35073063+smelchio@users.noreply.github.com>
